src/librustc_typeck/diagnostics.rs

```diff
@@ -152,6 +327,20 @@ fn(isize, *const *const u8) -> isize
 ```
 "##,
 
+E0166: r##"
+This error means that the compiler found a return expression in a function
+marked as diverging. A function diverges if it has `!` in the place of the
+return type in its signature. For example:
+
+```
+fn foo() -> ! { return; } // error
+```
+
+For a function that diverges, every control path in the function must end
+with a call to `panic!()` or another diverging function. Attempting to return
```

Comment:
Ahh, I didn't realize the thing with loops. (Also that sentence was paraphrased from the Reference, so I should probably update too).

Fixed.